lib/src/value/calculation.dart

```diff
@@ -223,6 +633,64 @@ class SassCalculation extends Value {
   /// simplification.
   SassCalculation._(this.name, this.arguments);
 
+  // Returns [value] coerced to [number]'s units.
+  static SassNumber _matchUnits(double value, SassNumber number) {
+    return SassNumber.withUnits(value, numeratorUnits: number.numeratorUnits);
+  }
+
+  /// Returns a rounded [number] based on a selected rounding [strategy],
+  /// to the nearest integer multiple of [step].
+  static SassNumber _roundWithStep(
+      String strategy, SassNumber number, SassNumber step) {
+    if (!{'nearest', 'up', 'down', 'to-zero'}.contains(strategy)) {
+      throw ArgumentError(
+          "$strategy must be either nearest, up, down or to-zero.");
+    }
+
+    if (number.value.isInfinite && step.value.isInfinite ||
+        step.value == 0 ||
+        number.value.isNaN ||
+        step.value.isNaN) {
+      return _matchUnits(double.nan, number);
+    }
+    if (number.value.isInfinite) return number;
+
+    if (step.value.isInfinite) {
+      return switch ((strategy, number.value)) {
+        (_, 0) => number,
+        ('nearest' || 'to-zero', >= 0) => _matchUnits(0.0, number),
```

Comment:
```suggestion
        ('nearest' || 'to-zero', > 0) => _matchUnits(0.0, number),
```

There's no reason for `>=` here (since you already checked `0` above), and a non-equal comparison matches the other cases.